src/extension.ts

```diff
@@ -364,8 +393,15 @@ export class Extension implements RunHooks {
     }
 
     try {
-      for (const model of this._models.enabledModels())
-        await this._runTest(this._testRun, include ? [...include] : [], testItemForGlobalErrors, new Set(), model, mode === 'debug', enqueuedTests.length === 1);
+      for (const model of this._models.enabledModels()) {
+        let globalSetupResult: reporterTypes.FullResult['status'] = 'passed';
+        if (model.canRunGlobalHooks('setup')) {
+          const testListener = this._errorReportingListener(this._testRun);
```

Comment:
Should we pass `testItemForGlobalErrors` here?

Reply:
I'll follow up.